sdk/core/azure-core/azure/core/utils/_utils.py

```diff
@@ -143,3 +159,7 @@ def __eq__(self, other: Any) -> bool:
 
     def __repr__(self) -> str:
         return str(dict(self.items()))
+
+    def items(self) -> _ItemsView:
+        """Return a new view of the dictionary's items."""
```

Comment:
Since this will now make it **the** behavior for everyone using it, is this something everyone would want, or should it be isolated to the original file?

Reply:
I think we do want this behaviour for everything for the same reason why we had this behaviour in the first place. I actually don't know what the original reason was, but the current `CaseInsensitiveDict` fills the same role as `_requests_basic._CaseInsensitiveDict`: storing request/response headers. Therefore, we want the same behaviour (I realize this isn't the greatest logic).

I talked @iscai-msft, she told me to ask @johanste or @annatisch who were the ones who original suggested this behaviour. Hopefully, they can provide some input.

Reply:
Yep it will be good to check with them. Right now, its within `aiohttp` & `requests basic`, which is a large part of our usage so it should be largely fine.